### PR TITLE
fix: move  channel parameter from sign in to sign up

### PR DESCRIPTION
**Pull request by J0**
The `channel` parameter for WhatsApp should be on the  `Signup` method instead of the `SignIn` method

Relevant Server side code: https://github.com/supabase/gotrue/blob/master/internal/api/signup.go#L22

### CODE REVIEW

File: src/lib/helpers.ts [hunk 1]
```diff
@@ -245,7 +245,9 @@ export function generatePKCEVerifier() {
   const verifierLength = 56
   const array = new Uint32Array(verifierLength)
   if (typeof window.crypto === 'undefined') {
-    throw new Error('PKCE is not supported on devices without WebCrypto API support or equivalent polyfills')
+    throw new Error(
+      'PKCE is not supported on devices without WebCrypto API support or equivalent polyfills'
```
Comment (hf) [inline src/lib/helpers.ts:249]:
```suggestion
      'PKCE is not supported on devices without WebCrypto API support, please add polyfills'
```

File: src/lib/helpers.ts [hunk 2]
```diff
@@ -255,7 +257,9 @@ async function sha256(randomString: string) {
   const encoder = new TextEncoder()
   const encodedData = encoder.encode(randomString)
   if (typeof window.crypto === 'undefined') {
-    throw new Error('PKCE is not supported on devices without WebCrypto API support or equivalent polyfills')
+    throw new Error(
+      'PKCE is not supported on devices without WebCrypto API support or equivalent polyfills'
```
Comment (hf) [inline src/lib/helpers.ts:261]:
```suggestion
      'PKCE is not supported on devices without WebCrypto API support, please add polyfills'
```

**Review by hf — APPROVED** [review]

**Comment by dshukertjr** [comment]
@J0 Do we need the `channel` parameter support for `resend()` method as well?

**Comment by J0** [comment]
@dshukertjr sorry just seeing this. Don't think we need it at the moment as it is not on the API. Let us know if customers request for it though and we can add it.

**Pull request merged**